lasagne/layers/recurrent.py

```diff
@@ -733,11 +780,16 @@ def step_masked(input_dot_W_n, mask_n, cell_previous, hid_previous,
             hid_init = T.dot(ones, self.hid_init)  # repeat num_batch times
 
         non_seqs = [self.W_hid_stacked]
-
         if self.peepholes:
             non_seqs += [self.W_cell_to_ingate,
                          self.W_cell_to_forgetgate,
                          self.W_cell_to_outgate]
+        else:
+            non_seqs += [(), (), ()]
```

Comment:
Sorry, but I'm confused as to why you need to even add these in at all - why would Theano want empty tuples for parameters it's not using?  In this case, for example, why does `non_seqs` need to be  `[self.W_hid_stacked, (), (), ()]` and not just `[self.W_hid_stacked]`?  Since you're giving `None` defaults to the keyword arguments ~line 702, the function should be callable without those (though I'm not sure why `step` has defaults and `step_masked` doesn't).